### PR TITLE
[UI]All Context Chips are visible in Dashboard

### LINKED ISSUE #5100: [UI] Current Context should be visible in the text field
#### Current Behavior
The Current Context Text field doesn't show the current context name. 

#### Desired Behavior
The Current context should be visible in the text field

#### Screenshots/Logs
Screenshot 2022-02-21 at 10 34 20 AM

#### Environment

- Browser: Chrome Safari Firefox
- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Harshit-Dandriyal**
Signed-off-by: Harshit Dandriyal <56961488+Harshit-Dandriyal@users.noreply.github.com>

**Description**
Kubernates Cluster UI support

Screenshot from 2022-04-28 15-06-57

Screenshot from 2022-04-30 10-03-23

This PR fixes #5100, #5099 , #5101

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
Can you please add screenshots of the changes??
Also, please remove unnecessary console log statements.

**Comment by Harshit-Dandriyal** [comment]
> Can you please add screenshots of the changes?? Also, please remove unnecessary console log statements
This PR is not done with some manual custom testing yet, I am in progress. I made this PR. So, I can get some suggestions from the community after getting a basic idea of the code

File: ui/components/DashboardComponent.js [hunk 1]
```diff
@@ -84,6 +84,9 @@ const styles = (theme) => ({
     height : "100%",
   },
 });
+async function fetchAllContexts(number) {
```
Comment (Abhishek-kumar09) [inline ui/components/DashboardComponent.js:87]: If this number signifies `pageSize` then please rename the variable to `pageSize`.

File: ui/components/DashboardComponent.js [hunk 2]
```diff
@@ -222,6 +232,10 @@ class DashboardComponent extends React.Component {
     this._isMounted = true
     this.fetchAvailableAdapters();
 
+    fetchAllContexts(25)
+      .then(res => this.setState({ contexts : res.contexts }))
```
Comment (Abhishek-kumar09) [inline ui/components/DashboardComponent.js:236]: probably add an extra check if contexts field is present in the res. We had few problems in the past handling undefined in such dynamic fetch calls.

File: ui/components/MeshConfigComponent.js [hunk 3]
```diff
@@ -686,10 +686,13 @@ handleNATSClick = () => {
           </ListItem>
         </List>
       );
+      { console.log(this.state) }
+      { console.log(this.props) }
```
Comment (Abhishek-kumar09) [inline ui/components/MeshConfigComponent.js:690]: You can remove this from the final commit.

File: ui/components/DashboardComponent.js [hunk 4]
```diff
@@ -758,35 +772,56 @@ class DashboardComponent extends React.Component {
...
+        //     ? "Using In Cluster Config"
+        //     : contextName}
+        //   onClick={self.handleKubernetesClick}
+        //   icon={<img src="/static/img/kubernetes.svg" className={classes.icon} />}
+        //   className={classes.chip}
+        //   key="k8s-key"
+        //   variant="outlined"
+        // />
+        <div>
+          {p(contextName)}
```
Comment (Abhishek-kumar09) [inline ui/components/DashboardComponent.js:803]: Please remove this `p`.

File: ui/components/DashboardComponent.js [hunk 5]
```diff
@@ -44,7 +44,7 @@ import { submitGrafanaConfigure } from "./GrafanaComponent";
 import { versionMapper } from "../utils/nameMapper";
 // podNameMapper,
 //import MesheryMetrics from "./MesheryMetrics";
-
+const p = console.log;
```
Comment (Abhishek-kumar09) [inline ui/components/DashboardComponent.js:47]:
```suggestion
```

**Comment by leecalcote** [comment]
> > Can you please add screenshots of the changes?? Also, please remove unnecessary console log statements
> 
> This PR is not done with some manual custom testing yet, I am in progress. I made this PR. So, I can get some suggestions from the community after getting a basic idea of the code, Adding some of the screenshots.

Use the “draft” label or “draft” setting the PR to indicate this.

File: ui/components/DashboardComponent.js [hunk 6]
```diff
@@ -137,24 +141,30 @@ class DashboardComponent extends React.Component {
 
   static getDerivedStateFromProps(props, state) {
     const {
-      meshAdapters, meshAdaptersts, k8sconfig, grafana, prometheus
+      //contextName
+      meshAdapters, meshAdaptersts, k8sconfig, grafana, prometheus, contextName
```
Comment (adithyaakrishna) [inline ui/components/DashboardComponent.js:145]:
```suggestion
      meshAdapters, meshAdaptersts, k8sconfig, grafana, prometheus, contextName
```

File: ui/components/DashboardComponent.js [hunk 7]
```diff
@@ -137,24 +141,30 @@ class DashboardComponent extends React.Component {
...
       st.inClusterConfig = k8sconfig.inClusterConfig;
       st.k8sfile = k8sconfig.k8sfile;
       st.contextName = k8sconfig.contextName;
       st.clusterConfigured = k8sconfig.clusterConfigured;
       st.configuredServer = k8sconfig.configuredServer;
       st.kts = props.ts;
+      if (!state.contextsFromFile?.length)
+        st.contextsFromFile = { ...st, contextsFromFile : [{ contextName, currentContext : true }] };
+      return st;
+    // }
```
Comment (adithyaakrishna) [inline ui/components/DashboardComponent.js:164]:
```suggestion
      return st;
```

File: ui/components/DashboardComponent.js [hunk 8]
```diff
@@ -31,7 +32,6 @@ import { withRouter } from "next/router";
 import { withSnackbar } from "notistack";
 import CloseIcon from "@material-ui/icons/Close";
 import { updateGrafanaConfig, updatePrometheusConfig, updateProgress } from "../lib/store";
-import dataFetch from "../lib/data-fetch";
 // import subscribeServiceMeshEvents from "./graphql/subscriptions/ServiceMeshSubscription";
 import subscribeDataPlaneEvents from "./graphql/subscriptions/DataPlanesSubscription";
```
Comment (adithyaakrishna) [inline ui/components/DashboardComponent.js:36]:
```suggestion
import subscribeDataPlaneEvents from "./graphql/subscriptions/DataPlanesSubscription";
```

File: ui/components/DashboardComponent.js [hunk 9]
```diff
@@ -44,7 +44,6 @@ import { submitGrafanaConfigure } from "./GrafanaComponent";
 import { versionMapper } from "../utils/nameMapper";
 // podNameMapper,
 //import MesheryMetrics from "./MesheryMetrics";
```
Comment (adithyaakrishna) [inline ui/components/DashboardComponent.js:46]:
```suggestion
import { versionMapper } from "../utils/nameMapper";
```

File: ui/components/DashboardComponent.js [hunk 10]
```diff
@@ -758,43 +766,56 @@ class DashboardComponent extends React.Component {
...
+        //   label={inClusterConfig
+        //     ? "Using In Cluster Config"
+        //     : contextName}
+        //   onClick={self.handleKubernetesClick}
+        //   icon={<img src="/static/img/kubernetes.svg" className={classes.icon} />}
+        //   className={classes.chip}
+        //   key="k8s-key"
+        //   variant="outlined"
+        // />
+        <div>
```
Comment (adithyaakrishna) [inline ui/components/DashboardComponent.js:795]:
```suggestion
        <div>
```

File: ui/components/MeshConfigComponent.js [hunk 11]
```diff
@@ -904,27 +928,43 @@ handleNATSClick = () => {
...
+                      <FormControl variant="outlined" fullWidth>
+                        <InputLabel id="select-label">Context Name</InputLabel>
+                        <Select
+                          id="contextName"
+                          name="contextName"
+                          labelId="select-label"
+                          label="Context Name"
+                          value={selectContext || contextName }
+                          onChange={this.selectcontext()}
+                          // disabled={inClusterConfigForm === true}
```
Comment (adithyaakrishna) [inline ui/components/MeshConfigComponent.js:942]:
```suggestion
```

**Comment by adithyaakrishna** [comment]
@Harshit-Dandriyal @Abhishek-kumar09  Is the lint failing or anything? Because I see a some empty lines between params and imports 😶

**Comment by Abhishek-kumar09** [comment]
> Is the lint failing or anything? Because I see some empty lines between params and imports no_mouth

No it is good. @Harshit-Dandriyal you can run linter by `npm run lint:fix`

**Comment by Harshit-Dandriyal** [comment]
> > Is the lint failing or anything? Because I see some empty lines between params and imports no_mouth
> 
> No it is good. @Harshit-Dandriyal you can run linter by `npm run lint:fix`

okayy

File: ui/components/DashboardComponent.js [hunk 12]
```diff
@@ -137,25 +140,26 @@ class DashboardComponent extends React.Component {
...
+    st.prometheus = prometheus;
     if (k8sconfig.ts > state.kts) {
       st.inClusterConfig = k8sconfig.inClusterConfig;
       st.k8sfile = k8sconfig.k8sfile;
       st.contextName = k8sconfig.contextName;
       st.clusterConfigured = k8sconfig.clusterConfigured;
       st.configuredServer = k8sconfig.configuredServer;
       st.kts = props.ts;
+      if (!state.contextsFromFile?.length)
+        st.contextsFromFile = { ...st, contextsFromFile : [{ contextName, currentContext : true }] };
```
Comment (utkarsh-pro) [inline ui/components/DashboardComponent.js:160]: Is this what you wanted to do?
```suggestion
        st.contextsFromFile = { ...(st.contextsFromFile || []), contextsFromFile : [{ contextName, currentContext : true }] };
```

File: ui/components/DashboardComponent.js (hunk 2 above)
Comment (utkarsh-pro) [reply ui/components/DashboardComponent.js:230]: Assuming `contexts` is an array

```suggestion
      .then(res => this.setState({ contexts : res?.contexts || [] }))
```

File: ui/components/MeshConfigComponent.js [hunk 13]
```diff
@@ -344,7 +358,16 @@ class MeshConfigComponent extends React.Component {
           })
       }
       if (name === "context") {
-        changeContext(event.target.value).
+        this.setState(() => {
+          return { contextName : this.state.selectContext }
+        })
+        let contextid;
+        this.state.contexts.forEach(element => {
+          if (element.name==this.state.selectContext){
```
Comment (utkarsh-pro) [inline ui/components/MeshConfigComponent.js:366]: Let's try to use `===` instead of `==`?

**Comment by utkarsh-pro** [comment]
@Harshit-Dandriyal, it's nice to have you here 🎉 . Meshery can't have too many of C11/12/13 folks, can it? 😉

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal, it's nice to have you here tada . Meshery can't have too many of C11/12/13 folks, can it? wink

:joy: :joy: , Abhishek recommended this place to hangout :smile: , Also Environment here is soothing, learning about service meshes, Kubernetes, and some advance REACT used in industry instead of plain JS(FreeLancing :wink: )

File: ui/components/MeshConfigComponent.js (hunk 13 above)
Comment (Harshit-Dandriyal) [reply ui/components/MeshConfigComponent.js:366]: Yup, Let's go :smile:

File: ui/components/DashboardComponent.js (hunk 12 above)
Comment (Harshit-Dandriyal) [reply ui/components/DashboardComponent.js:160]: :thinking:, ..yes

**Comment by Harshit-Dandriyal** [comment]
Thank you @utkarsh-pro, @adithyaakrishna and @Abhishek-kumar09 :smile:

**Comment by leecalcote** [comment]
@Harshit-Dandriyal will you resolve any open comments that have been addressed?

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal will you resolve any open comments that have been addressed?

@leecalcote All the suggested changes are commited and pushed, Code is working as expected 🤗

**Comment by leecalcote** [comment]
Thanks. Here's the button that I'm asking you to click when comments have been addressed:
Screen Shot 2022-05-02 at 8 59 45 AM

**Comment by Harshit-Dandriyal** [comment]
> Thanks. Here's the button that I'm asking you to click when comments have been addressed: Screen Shot 2022-05-02 at 8 59 45 AM

Okay, Got it for next time :smile:

**Pull request merged**